src/Diagnostics/FxCop/SDL.Analyzer.Common/CSharp/Shared/SyntaxNodeHelper.cs

```diff
@@ -0,0 +1,323 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace SDL.Analyzer.Common
+{
+    public sealed class CSharpSyntaxNodeHelper : SyntaxNodeHelper
+    {
+        private static CSharpSyntaxNodeHelper instance = new CSharpSyntaxNodeHelper();
+
+        public static CSharpSyntaxNodeHelper Default { get { return instance; } }
+
+        private CSharpSyntaxNodeHelper()
+        {
+        }
+
+        public override bool ContainsMethodCall(SyntaxNode node, Func<string, bool> predicate)
```

Comment:
Again, this helper should be deleted. Right approach is to right a code block analyzer, and use a bool state flag to track if any invocation expression was encountered, and if so do the relevant analysis in CodeBlockEndAction.